web/src/pages/billing/BillingCostByTime.tsx

```diff
@@ -51,182 +72,480 @@ const BillingCostByTime: React.FunctionComponent = () => {
-        }
 
-        // Handle columns parameter
-        if (columns !== undefined && columns.size > 0) {
-            const columnsArray = Array.from(columns).sort()
-            searchParams.set('columns', columnsArray.join(','))
-        }
+            // Handle projects parameter
+            if (projects !== undefined) {
+                if (projects.length > 0) {
+                    searchParams.set('projects', projects.join(','))
+                } else {
+                    searchParams.delete('projects')
+                }
+            }
 
-        const url = generateUrl(location, {
-            groupBy: grp,
-            selectedData: selData,
-            start: st,
-            end: ed,
-            expand: searchParams.get('expand') || undefined,
-            columns: searchParams.get('columns') || undefined,
-        })
-        navigate(url)
-    }
+            // Handle topics parameter
+            if (topics !== undefined) {
+                if (topics.length > 0) {
+                    searchParams.set('topics', topics.join(','))
+                } else {
+                    searchParams.delete('topics')
+                }
+            }
+
+            const url = generateUrl(location, {
+                groupBy: grp,
+                selectedData: selData,
+                start: st,
+                end: ed,
+                expand: searchParams.get('expand') || undefined,
+                columns: searchParams.get('columns') || undefined,
+                projects: searchParams.get('projects') || undefined,
+                topics: searchParams.get('topics') || undefined,
+            })
+            navigate(url)
+        },
+        [location, navigate]
+    )
 
     const onGroupBySelect = (event: unknown, data: DropdownProps) => {
         const value = data.value
         if (typeof value == 'string') {
             setGroupBy(value as BillingColumn)
             setSelectedData(undefined)
-            updateNav(value, undefined, start, end, expandCompute, visibleColumns)
+            // Clear filters when switching groupBy
+            if (value !== BillingColumn.GcpProject) {
+                setSelectedProjects([])
+            }
+            if (value !== BillingColumn.Topic) {
+                setSelectedTopics([])
+            }
+            updateNav(value, undefined, start, end, expandCompute, visibleColumns, [], [])
         }
     }
 
     const onSelect = (event: unknown, data: DropdownProps) => {
         const value = data.value
         if (typeof value == 'string') {
             setSelectedData(value)
-            updateNav(groupBy, value, start, end, expandCompute, visibleColumns)
+            updateNav(
+                groupBy,
+                value,
+                start,
+                end,
+                expandCompute,
+                visibleColumns,
+                selectedProjects,
+                selectedTopics
+            )
         }
     }
 
+    const onProjectsSelect = (
```

Comment:
I'm not sure exactly where the bug is, but filtering by project/topic doesn't seem to be working at the moment, it updates the URL and if you then refresh the page the data updates - but new data isn't fetched when you select topics/projects